### PR TITLE
hw/mcu/dialog: Add helper to enable/disable AMBA clocks

### PR DESCRIPTION
**Pull request by wes3**
Add a helper to enable/disable AMBA clocks in CRG_TOP->CLK_AMBA_REG.
The code was doing an un-protected read-modify-write and if the clock is modified inside ISR's this could cause unpredictable behavior.

The helper will enable/disable interrupts around the setting/clearing of bits in that register. The helpers added are:
da1469x_clock_amba_enable(mask)
da1469x_clock_amba_disable(mask)

### CODE REVIEW

File: hw/mcu/dialog/da1469x/include/mcu/da1469x_clock.h [hunk 1]
```diff
@@ -94,6 +95,36 @@ uint32_t da1469x_clock_lp_rcx_freq_get(void);
  */
 void da1469x_clock_lp_rcx_disable(void);
 
+/**
+ * Enable an AMBA clock
+ *
+ * @param idx
```
Comment (kasjer) [inline hw/mcu/dialog/da1469x/include/mcu/da1469x_clock.h]: idx or mask?

Comment (wes3) [reply hw/mcu/dialog/da1469x/include/mcu/da1469x_clock.h]: mask :-)

File: hw/mcu/dialog/da1469x/include/mcu/da1469x_hal.h [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #define __MCU_DA1469X_HAL_H_
 
 #include <assert.h>
+#include "mcu/cmsis_nvic.h"
```
Comment (kasjer) [inline hw/mcu/dialog/da1469x/include/mcu/da1469x_hal.h]: cmsis_nvic.h seems to be unrelated to change

Comment (wes3) [reply hw/mcu/dialog/da1469x/include/mcu/da1469x_hal.h]: I guess you can say it is unrelated. Looks like cmsis_nvic.h brings in mcu.h which is needed for files that include da1469x_hal.h. I will change it to mcu.h
